tests/ticket2707.py

```diff
@@ -64,11 +70,57 @@ def tearDown(self):
 
     def testMatchXy(self):
         matches = afwTable.matchXy(self.cat1, self.cat2, 0.01)
-        self.assertEquals(len(matches), 6)
+        self.assertEquals(len(matches), self.nUniqueMatch)
+
         for m in matches:
-            self.assertEquals(m.first.getId() + 10, m.second.getId())
+            self.assertEquals(m.first.getId() + self.nobj, m.second.getId())
             self.assertEquals(m.distance, 0.0)
 
+    def testMatchXyMatchControl(self):
+        """Test using MatchControl to return all matches, and add a test for closest==False at the same time"""
+        for closest in (True, False):
+            for includeMismatches in (True, False):
+                mc = afwTable.MatchControl()
+                mc.findOnlyClosest = closest
+                mc.includeMismatches = includeMismatches
+                matches = afwTable.matchXy(self.cat1, self.cat2, 0.01, mc)
+
+                if False:
+                    for m in matches:
+                        print closest, m.first.getId(), m.second.getId(), m.distance
+
+                if includeMismatches:
+                    catMatches = afwTable.SourceCatalog(self.table)
+                    catMismatches = afwTable.SourceCatalog(self.table)
+                    for m in matches:
+                        if m[1] != None:
+                            if not any(x == m[0] for x in catMatches):
+                                catMatches.append(m[0])
+                        else:
+                            catMismatches.append(m[0])
+                    matches = afwTable.matchXy(catMatches, self.cat2, 0.01, mc)
+                    mc.includeMismatches = False
+                    noMatches = afwTable.matchXy(catMismatches, self.cat2, 0.01, mc)
+                    self.assertEquals(len(noMatches), 0)
```

Comment:
Changed some uses of `assertEquals` but left others because it will be annoying to change them in original commits.  Looking at the docs for `TestCase`, `assertEquals` is perfectly legitimate, and it appears to be fairly widely used in afw.